src/lib/profiles/software-update/SoftwareUpdateProfile.cpp

```diff
@@ -28,460 +28,575 @@
- * returns: error/status
+
+/**
+ * Serialize the object to the provided MessageIterator
+ *
+ * @param[in] i An iterator over the message being packed
+ *
+ * @retval WEAVE_NO_ERROR  On success.
+ * @retval WEAVE_ERROR_BUFFER_TOO_SMALL If the list is too long to fit in the message.
  */
-WEAVE_ERROR IntegrityTypeList::pack(MessageIterator &i)
+WEAVE_ERROR IntegrityTypeList::pack(MessageIterator & i)
 {
-  if (i.hasRoom(theLength+1)) {
-    i.writeByte(theLength);
-    for (int j=0; j<theLength; j++) i.writeByte(theList[j]);
-    return WEAVE_NO_ERROR;
-  }
-  else
-    return WEAVE_ERROR_INVALID_LIST_LENGTH;
+    if (i.hasRoom(theLength + 1))
+    {
+        i.writeByte(theLength);
+        for (int j = 0; j < theLength; j++)
+            i.writeByte(theList[j]);
+        return WEAVE_NO_ERROR;
+    }
+    else
+        return WEAVE_ERROR_BUFFER_TOO_SMALL;
 }
-/*
- * parameters:
- * - MessageIterator &i, an iterator over the message being parsed
- * - IntegrityTypeList &aList, a pointer to an object to contain the result
- * returns: error/status
+
+/**
+ * Deserialize the object from the given MessageIterator into provided IntegrityTypeList
+ *
+ * @param[in] i An iterator over the message being parsed.
+ * @param[in] aList A reference to an object to contain the result
+ *
+ * @retval WEAVE_NO_ERROR  On success.
+ * @retval WEAVE_ERROR_BUFFER_TOO_SMALL Message was too short.
+ * @retval WEAVE_ERROR_INVALID_LIST_LENGTH If the message contained an invalid list length (either not enough data to fill in the
+ * list or too many to fit within the limits)
  */
-WEAVE_ERROR IntegrityTypeList::parse(MessageIterator &i, IntegrityTypeList &aList)
+WEAVE_ERROR IntegrityTypeList::parse(MessageIterator & i, IntegrityTypeList & aList)
 {
-  i.readByte(&aList.theLength);
-  if (!i.hasData(aList.theLength)) return WEAVE_ERROR_INVALID_LIST_LENGTH;
-  for (int j=0; j<aList.theLength; j++) i.readByte(&aList.theList[j]);
-  return WEAVE_NO_ERROR;
+    TRY(i.readByte(&aList.theLength));
+    if (!i.hasData(aList.theLength))
+        return WEAVE_ERROR_BUFFER_TOO_SMALL;
+    if (aList.theLength > 3)
+        return WEAVE_ERROR_INVALID_LIST_LENGTH;
+
+    for (int j = 0; j < aList.theLength; j++)
+        i.readByte(&aList.theList[j]);
+    return WEAVE_NO_ERROR;
 }
-/*
- * parameter: IntegrityTypeList &another, another list to check against
- * returns: true if the lists are equal, false otherwise
+
+/**
+ * An equality operator
+ *
+ * @param another A list to check against this list
+ * @return true if the lists are equal, false otherwise
  */
-bool IntegrityTypeList::operator == (const IntegrityTypeList &another) const
+bool IntegrityTypeList::operator ==(const IntegrityTypeList & another) const
 {
-  if (theLength!=another.theLength)
-    return false;
-  for (int i=0; i<theLength; i++) {
-    if (theList[i]!=another.theList[i])
-      return false;
-  }
-  return true;
+    if (theLength != another.theLength)
```

Comment:
Would favor a single return path.

Reply:
No opposition from me if there are future refactorings in the works.

Reply:
I folded and refactored already. :)